Promise/Promise+Extras.swift

```diff
@@ -73,19 +99,30 @@ public enum Promises {
             }).then(fulfill).catch(reject)
         })
     }
-
+    
+    /// Runs a block of code that creates a `Promise` but which can also `throw`.
+    /// - Parameter block: Code to generate a `Promise`; `throw`s are rejected as expcted.
+    /// - Returns: `Promise<T>` that is created from the given block
     public static func kickoff<T>(_ block: @escaping () throws -> Promise<T>) -> Promise<T> {
         return Promise(value: ()).then(block)
     }
 
+    /// Allows for a block of code that could `throw` to be executed and turned into a `Promise`.
+    /// - Parameter block: Code to be executed; `throw`s are rejected as expected.
+    /// - Returns: `Promise<T>` that executes the given block, and rejects if necessary.
```

Comment:
```suggestion
    /// Runs a block of code that returns a `Promise` but which can also `throw`.
    /// Useful when setting up a promise with code that needs to throw.
    /// - Parameter block: Code to generate a value that will be wrapped in the promise; `throw`ing creates a rejected promise with the thrown error.
    /// - Returns: `Promise<T>` that executes the given block, and rejects if necessary.
```

Reply:
@khanlou this overload's closure doesn't return a `Promise`; it returns an instance of `Value`... unless I'm missing something.

Reply:
Yeah, I think I goofed the copy paste on this one. Let's go with `returns a value but`?